contracts/AjnaKeeperTaker.sol

```diff
@@ -0,0 +1,194 @@
+
+    /// @dev Actor allowed to take auctions using this contract
+    address public immutable owner;
+
+    /// @dev Identifies the Ajna deployment, used to validate pools
+    PoolDeployer public immutable poolFactory;
+
+
+    // sig: 0x2083cd40
+    /// @notice Pool invoking callback is not from the Ajna deployment configured in this contract.
+    error InvalidPool();
+
+    // sig: 0x82b42900
+    /// @notice Caller is not the owner of this contract.
+    error Unauthorized();
+
+    // sig: 0xf54a7ed9
+    /// @notice Emitted when the requested liquidity source is not available on this deployment of the contract.
+    error UnsupportedLiquiditySource();
+
+
+    /// @param ajnaErc20PoolFactory Ajna ERC20 pool factory for the deployment of Ajna the keeper is interacting with.
+    constructor(PoolDeployer ajnaErc20PoolFactory) {
+        owner = msg.sender;
+        poolFactory = ajnaErc20PoolFactory;
+    }
+
+    /// @notice Owner may call to recover legitimate ERC20 tokens sent to this contract.
+    function recover(IERC20 token) public onlyOwner {
+        uint256 balance = token.balanceOf(address(this));
+        token.transfer(owner, balance);
+    }
+
+    /// @notice Called by keeper to invoke `Pool.take`, passing `IERC20Taker` callback data.
+    /// @param pool ERC20 pool with an active auction.
+    /// @param borrowerAddress Identifies the liquidation to take.
+    /// @param auctionPrice Last known price of the auction, in `WAD` precision, used for quote token approval.
+    /// @param maxAmount Limit collateral to take from the auction, in `WAD` precision.
+    /// @param source Identifies the source of liquidity to use for the swap (e.g. 1inch).
+    /// @param swapRouter Address of the router to use for the swap.
+    /// @param swapDetails Source-specific data needed to perform the swap, which may be populated by an external API.
+    function takeWithAtomicSwap(
+        IERC20Pool pool,
+        address borrowerAddress,
+        uint256 auctionPrice,
+        uint256 maxAmount,
+        LiquiditySource source,
+        address swapRouter,
+        bytes calldata swapDetails
+    ) external onlyOwner {
+        // configuration passed through to the callback function instructing this contract how to swap
+        bytes memory data = abi.encode(
+            SwapData({
+                source: source,
+                router: swapRouter,
+                details: swapDetails
+            })
+        );
+
+        // approve the pool to spend this contract's quote token
+        uint256 approvalAmount = _ceilWmul(maxAmount, auctionPrice) / pool.quoteTokenScale(); // convert WAD to token precision
+        IERC20(pool.quoteTokenAddress()).approve(address(pool), approvalAmount);
+
+        // invoke the take
+        pool.take(borrowerAddress, maxAmount, address(this), data);
+
+        recover(IERC20(pool.quoteTokenAddress())); // send excess quote token (profit) to owner
+    }
```

Comment:
Is there a possibility we would want a `takeWithoutAtomicSwap` that would recover collateral to the owner, i.e. in the event that there would be too much slippage?

Reply:
We actually don't have a regular take which uses liquidity from the EOA.  Would be a nice-to-have feature, but not in scope for this PR.